lib/rdoc/code_object/class_module.rb

```diff
@@ -379,10 +394,10 @@ def marshal_load(array) # :nodoc:
 
     @comment    = RDoc::Comment.from_document document
 
-    @comment_location = if RDoc::Markup::Document === document.parts.first then
-                          document
+    @comment_location = if document.parts.first.is_a?(RDoc::Markup::Document)
+                          document.parts.map { |doc| [doc, doc.file] }
                         else
-                          RDoc::Markup::Document.new document
+                          [[document, document.file]]
```

Comment:
We were creating a brand new document before. Was that intentional?

For example, should the updated code be this?

```ruby
new_doc = RDoc::Markup::Document.new document
[[new_doc, new_doc.file]]
```